### PR TITLE
Support for synthetics

### PR DESCRIPTION
**Pull request by sriumcp**
Signed-off-by: Srinivasan Parthasarathy <spartha@us.ibm.com>

Addresses the etc3 portions of #iter8-tools/iter8#742 and #iter8-tools/iter8#751

### CODE REVIEW

**Review by huang195 — COMMENTED** [review]
Is the term `level` commonly used? Would it be more clear to call it `value`, so it takes the guessing work out of what it means when seeing `mock` and `value` together?

**Comment by huang195** [comment]
Neither is really clear, and I can't think of anything that is better at the moment. I guess it needs to be documented well.

File: api/v2alpha2/metric_types.go [hunk 1]
```diff
@@ -133,7 +144,16 @@ type MetricSpec struct {
 	// However, as indicated by its name, URLTemplate may be templated.
 	// In this case, Iter8 will attempt to substitute placeholders in the URLTemplate at query time using Secret.
 	// Placeholder substitution will be attempted only when Secret != nil.
-	URLTemplate string `json:"urlTemplate" yaml:"urlTemplate"`
+	// +optional
+	URLTemplate *string `json:"urlTemplate,omitempty" yaml:"urlTemplate,omitempty"`
+
+	// Mock enables mocking of metric values, which is useful in tests and tutorial/documentation.
+	// For a counter metric, the level is used as a rate. Roughly speaking,
```
[the remark this answers is not in the document]
Comment (kalantar) [reply api/v2alpha2/metric_types.go:151]: I see it clearly stated above in the PR conversation. Please add it to the comment as well.

**Review by kalantar — CHANGES_REQUESTED** [review]
one minor change requested

File: api/v2alpha2/metric_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sriumcp) [reply api/v2alpha2/metric_types.go]: Done!

**Review by kalantar — APPROVED** [review]
lgtm

**Pull request merged**